tensorflow/core/kernels/mkl/mkl_conv_ops.cc

```diff
@@ -1556,77 +1580,236 @@ class MklQuantizedConv2DOp
     }
   }
 
-  explicit MklQuantizedConv2DOp(OpKernelConstruction* context)
-      : MklConvOp<Device, Tinput, qint8, Tbias, Toutput, Ttemp_output, int32,
-                  bias_enabled, false, is_depthwise, native_format>(context) {
-    bool is_filter_const;
-    OP_REQUIRES_OK(context,
-                   context->GetAttr("is_filter_const", &is_filter_const));
+  explicit MklQuantizedConvOp(OpKernelConstruction* context)
+      : MklConvOp<Device, Tinput, /*Tfilter*/ qint8, Tbias, Toutput,
+                  Ttemp_output, /*Tpadding*/ int32,
+                  /*bias_enabled*/ false, /*pad_enabled*/ false, is_depthwise,
+                  /*native_format*/ true>(context) {
+    std::vector<std::vector<string>> supported_fusions = {
+        {"BiasAdd"},
+        {"Relu"},
+        {"Requantize"},
+        {"BiasAdd", "Relu"},
+        {"BiasAdd", "Requantize"},
+        {"Relu", "Requantize"},
+        {"BiasAdd", "Relu", "Requantize"},
+        {"BiasAdd", "Sum", "Relu"},
+        {"BiasAdd", "Sum", "Relu", "Requantize"}};
+
+    std::vector<string> fused_ops_attr;
+    // Old quantized ops don't have fused_ops attribute
+    if (context->HasAttr("fused_ops")) {
+      OP_REQUIRES_OK(context, context->GetAttr("fused_ops", &fused_ops_attr));
+    }
 
-    if (bias_enabled) {
+    // Number of fused ops for new API is determined by size of fused_ops_attr.
+    // For old API, num_fused_ops is used to determine number of fused ops.
+    // TODO(intel-tf): num_fused_ops and legacy_fused_ops should go away once
+    // old API is abandoned.
+    OP_REQUIRES(context, !(fused_ops_attr.size() > 0 && num_fused_ops > 0),
+                errors::InvalidArgument(
+                    "QuantizedConv fused ops should be only availabe through "
+                    "either new API or old API, got both."));
+
+    if (fused_ops_attr.size() > 0) {
+      fused_ops_ = fused_ops_attr;
+    } else if (num_fused_ops > 0) {
+      for (int i = 0; i < num_fused_ops; ++i) {
+        fused_ops_.push_back(legacy_fused_ops[i]);
+      }
+    }
+
+    if (fused_ops_.size() > 0) {
+      bool is_fusion_supported =
+          std::find(supported_fusions.begin(), supported_fusions.end(),
+                    fused_ops_) != supported_fusions.end();
+      OP_REQUIRES(context, is_fusion_supported,
+                  errors::InvalidArgument("Unsupported QuantizedConv fusion: [",
+                                          absl::StrJoin(fused_ops_, ","), "]"));
+    }
+
+    bool fuse_bias = std::find(fused_ops_.begin(), fused_ops_.end(),
+                               "BiasAdd") != fused_ops_.end();
+    DataType bias_dt, summand_dt, out_dt;
+    if (fuse_bias) {
+      this->set_fuse_biasadd(true);
```

Comment:
I think we can process this in a more concise way. 
Can we have an enum for all available fused ops, e.g..,
```c++
enum class oneDNNFusedOps {
  kBias = 0,
  kSum = 1,
  ...
};
```
And have an int32t or int64t variable storing binary flags for all those? This way we can just loop through `fused_ops_` once to mark all flags. 
```c++
int64t fused_op_flags_ = 0;  // Class member
// Process fused ops list
for (const auto& op : fused_ops) {
  fused_op_flags_ ^= 1 << StrToEnum[op];
}
// Convenience function to check if op is in fused ops, e.g., isFused(kBias).
inline bool isFused(oneDNNFusedOps op) {
  return fused_op_flags & (1 << op); 
}
```

Reply:
Done. Thanks for the detailed suggestion.